test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -77,9 +126,111 @@ expectResponseCode
+addressPoolGap =
+    lens _get _set
+  where
+    _get :: HasType (ApiT AddressPoolGap) s => s -> Word8
+    _get = getAddressPoolGap . getApiT . view typed
+    _set :: HasType (ApiT AddressPoolGap) s => (s, Word8) -> s
+    _set (s, v) = set typed (ApiT $ unsafeMkAddressPoolGap v) s
+
+balance :: HasType (ApiT WalletBalance) s => Lens' s WalletBalance
+balance =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> WalletBalance
+    _get = getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, WalletBalance) -> s
+    _set (s, v) = set typed (ApiT v) s
+
+balanceAvailable :: HasType (ApiT WalletBalance) s => Lens' s (Quantity "lovelace" Natural)
+balanceAvailable =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> Quantity "lovelace" Natural
+    _get = available . getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, Quantity "lovelace" Natural) -> s
+    _set (s, v) = set typed (ApiT $ WalletBalance {available = v, total = v }) s
+
+balanceTotal :: HasType (ApiT WalletBalance) s => Lens' s (Quantity "lovelace" Natural)
+balanceTotal =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> Quantity "lovelace" Natural
+    _get = total . getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, Quantity "lovelace" Natural) -> s
+    _set (s, v) = set typed (ApiT $ WalletBalance {available = v, total = v }) s
+
+delegation :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => Lens' s (WalletDelegation (ApiT PoolId))
+delegation =
+    lens _get _set
+  where
+    _get :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => s -> (WalletDelegation (ApiT PoolId))
+    _get = getApiT . view typed
+    _set :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => (s, (WalletDelegation (ApiT PoolId))) -> s
+    _set (s, v) = set typed (ApiT v ) s
+
+passphraseLastUpdate :: HasType (ApiT WalletPassphraseInfo) s => Lens' s UTCTime
+passphraseLastUpdate =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletPassphraseInfo) s => s -> UTCTime
+    _get = lastUpdatedAt . getApiT . view typed
+    _set :: HasType (ApiT WalletPassphraseInfo) s => (s, UTCTime) -> s
+    _set (s, v) = set typed (ApiT $ WalletPassphraseInfo v ) s
+
+state :: HasType (ApiT WalletState) s => Lens' s WalletState
+state =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletState) s => s -> WalletState
+    _get = getApiT . view typed
+    _set :: HasType (ApiT WalletState) s => (s, WalletState) -> s
+    _set (s, v) = set typed (ApiT v ) s
+
+walletName :: HasType (ApiT WalletName) s => Lens' s Text
+walletName =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletName) s => s -> Text
+    _get = getWalletName . getApiT . view typed
+    _set :: HasType (ApiT WalletName) s => (s, Text) -> s
+    _set (s, v) = set typed (ApiT $ WalletName v) s
+
+walletId :: HasType (ApiT WalletId) s => Lens' s (Digest Blake2b_160)
```

Comment:
Would it make sense maybe to go all the way to `Lens' s Text` for this one? Not sure how you'll generate a `Digest` easily in the scenario ^^"